lib/components/session/login.vue

```diff
@@ -0,0 +1,79 @@
+<!--
+Copyright 2017 Super Adventure Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/nafundi/super-adventure/blob/master/NOTICE.
+
+This file is part of Super Adventure. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of Super Adventure,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <div>
+    <heading title="Log in"/>
+    <alert type="danger" :message="error"/>
+    <form @submit.prevent="logIn">
+      <div class="form-group">
+        <label for="email">Email address</label>
+        <input type="email" v-model="email" ref="email" id="email"
+          class="form-control" placeholder="Email" required>
+      </div>
+      <div class="form-group">
+        <label for="password">Password</label>
+        <input type="password" v-model="password" id="password"
+          class="form-control" placeholder="Password" required>
+      </div>
+      <button type="submit" class="btn btn-primary" :disabled="disabled">
+        Log in
+      </button>
+    </form>
+  </div>
+</template>
+
+<script>
+import axios from 'axios';
+
+import { safeBack } from '../../router';
+
+export default {
+  data() {
+    return {
+      email: '',
+      password: '',
+      error: null,
+      disabled: false
+    };
+  },
+  mounted() {
+    this.$refs.email.focus();
```

Comment:
is there a way to generalize this for our forms so the first `input` is `focus()`ed? then we'll never forget to do it.

Reply:
I definitely think we should focus the first `input` when the user is creating a new resource. Do you think we should focus the first `input` when the user is editing an existing resource? Part of me feels like when editing, the user should always have to explicitly select a field before it becomes editable (rather than the first field starting out as editable). That way, if the user accidentally types something, it won't go in the first field. But maybe that concern's overblown, and it'd definitely make implementation trickier...

Reply:
no; focusing the first field is important for accessibility reasons if nothing else. it's in the criteria document, too.